server/controllers/users/signup.js

```diff
@@ -0,0 +1,24 @@
+const { hash } = require('bcrypt');
+const agentSchema = require('../../utils/validation/agentSchema');
+const signUpQuery = require('../../database/quieres/account/signUp');
+const { signToken } = require('../../utils');
+
+module.exports = async (req, res, next) => {
+  try {
+    const {
+      error, value: {
+        password, email, username, phone,
+      },
+    } = agentSchema.validate(req.body);
+    if (error) return res.status(400).json({ message: error.details[0].message });
+    const hasedPasword = await hash(password, 10);
+    await signUpQuery(username, email, phone, hasedPasword);
+    const token = await signToken(email, username, phone);
+    return res.status(201).cookie('token', token).json({ message: 'user created' });
+  } catch (err) {
+    if (err.code === '23505') {
+      return res.status(400).json({ message: 'username or phone already exists' });
```

Comment:
This message doesn't describe every thing since this could happen due to the existence of the username, email or phone so you should describe the error in a better way or make it public like: 'The user is already exists'.